src/sidebar/components/sidebar-content-omega.js

```diff
@@ -0,0 +1,168 @@
+  const linkedAnnotationId = useStore(store =>
+    store.directLinkedAnnotationId()
+  );
+  const linkedAnnotation = useStore(store =>
+    store.findAnnotationByID(linkedAnnotationId)
+  );
+  const searchUris = useStore(store => store.searchUris());
+  const sidebarHasOpened = useStore(store => store.hasSidebarOpened());
+  const userId = useStore(store => store.profile().userid);
+
+  // The local `$tag` of a direct-linked annotation; populated once it
+  // has anchored: meaning that it's ready to be focused and scrolled to
+  const linkedAnnotationAnchorTag =
+    linkedAnnotation && linkedAnnotation.$orphan === false
+      ? linkedAnnotation.$tag
+      : null;
+
+  // Actions
+  const clearSelectedAnnotations = useStore(
+    store => store.clearSelectedAnnotations
+  );
+  const selectTab = useStore(store => store.selectTab);
+
+  const hasDirectLinkedAnnotationError = useStore(store => {
+    if (isLoading || !linkedAnnotationId) {
+      return false;
+    }
+    return !store.annotationExists(linkedAnnotationId);
+  });
+
+  const hasDirectLinkedGroupError = useStore(store =>
+    store.directLinkedGroupFetchFailed()
+  );
+
+  const hasContentError =
+    hasDirectLinkedAnnotationError || hasDirectLinkedGroupError;
+
+  const showTabs = !hasContentError && !hasAppliedFilter;
+  const showSearchStatus = !hasContentError && !isLoading;
+
+  // Show a CTA to log in if successfully viewing a direct-linked annotation
+  // and not logged in
+  const showLoggedOutMessage = (() => {
+    if (
+      linkedAnnotationId &&
+      !isLoggedIn &&
+      !hasDirectLinkedAnnotationError &&
+      !isLoading
+    ) {
+      return true;
+    }
+    return false;
+  })();
+
+  const prevGroupId = useRef(focusedGroupId);
+
+  // Reload annotations when group, user or document search URIs change
+  useEffect(() => {
+    if (!prevGroupId.current || prevGroupId.current !== focusedGroupId) {
+      // Clear any selected annotations when the group ID changes
+      clearSelectedAnnotations();
+      prevGroupId.current = focusedGroupId;
+    }
+    if (focusedGroupId && searchUris.length) {
+      loadAnnotationsService.load(searchUris, focusedGroupId);
+    }
+  }, [
+    clearSelectedAnnotations,
+    loadAnnotationsService,
+    focusedGroupId,
+    userId,
+    searchUris,
+  ]);
+
+  // When a `linkedAnnotationAnchorTag` becomes available, scroll to it
+  // and focus it
+  useEffect(() => {
+    if (linkedAnnotationAnchorTag) {
+      frameSync.focusAnnotations([linkedAnnotationAnchorTag]);
+      frameSync.scrollToAnnotation(linkedAnnotationAnchorTag);
+      selectTab(tabForAnnotation(linkedAnnotation));
+    }
+  }, [frameSync, linkedAnnotation, linkedAnnotationAnchorTag, selectTab]);
+
+  // Connect to the streamer when the sidebar has opeend or if user is logged in
+  useEffect(() => {
+    if (sidebarHasOpened || isLoggedIn) {
+      streamer.connect();
+    }
+  }, [streamer, sidebarHasOpened, isLoggedIn]);
```

Comment:
I want to verify that this isn't overzealous about connecting. It doesn't seem to fire too many times in local test for me when changing users.

Reply:
`streamer.connect()` does nothing if already connected. This worked as I expected in my testing. It didn't connect after startup if the sidebar was closed and the user not logged in, but did connect otherwise.